internal/controller/vrg_volrep.go

```diff
@@ -1592,35 +1601,49 @@ func (v *VRGInstance) checkResyncCompletionAsSecondary(pvc *corev1.PersistentVol
 	return true
 }
 
-// isVRConditionMet returns true if the condition is met, and an error mesage if we could not get the
-// condition value.
+// isVRConditionMet check if condition is met.
+// Returns 3 values:
+//   - met: true if the condition status matches the desired status, otherwise false
+//   - ok: true if the check was successful; the condition was found, its observedGeneration matches the object
+//     generation, and its value is not unknown.
+//   - errorMsg: error message describing why the condition is not met
 func isVRConditionMet(volRep *volrep.VolumeReplication,
 	conditionType string,
 	desiredStatus metav1.ConditionStatus,
-) (bool, string) {
+) (bool, bool, string) {
+	met := true
+	ok := true
+
 	volRepCondition := findCondition(volRep.Status.Conditions, conditionType)
 	if volRepCondition == nil {
 		errorMsg := fmt.Sprintf("Failed to get the %s condition from status of VolumeReplication resource.",
 			conditionType)
 
-		return false, errorMsg
+		return !met, !ok, errorMsg
 	}
 
 	if volRep.GetGeneration() != volRepCondition.ObservedGeneration {
 		errorMsg := fmt.Sprintf("Stale generation for condition %s from status of VolumeReplication resource.",
 			conditionType)
 
-		return false, errorMsg
+		return !met, !ok, errorMsg
 	}
 
 	if volRepCondition.Status == metav1.ConditionUnknown {
 		errorMsg := fmt.Sprintf("Unknown status for condition %s from status of VolumeReplication resource.",
 			conditionType)
 
-		return false, errorMsg
+		return !met, !ok, errorMsg
+	}
+
+	// We can compare the condition status.
+
+	if volRepCondition.Status != desiredStatus {
+		// csi-addons > 0.10.0 returns defailed error message
```

Comment:
returns what? `defailed` is a new word to me :)

Reply:
detailed :-)

Reply:
fixed in current version.